### ISSUE TITLE
ArcFace option to return logits to get metrics during training

### ISSUE THREAD

**Issue by bsridatta** [KevinMusgrave/pytorch-metric-learning#446]
Since ArcFace uses softmax and cross-entropy loss, there is the possibility to get accuracy during training and eval with this. But the accuracy calculators take embeddings after training. 

It would be great to have an option in ArcFace loss that also returns logits. There is the `get_logit` method but again it takes embeddings as input making the redundant calculations.

**Comment by KevinMusgrave** [comment]
I don't fully understand. Can you write a code example of how you'd like it to work?

**Comment by bsridatta** [comment]
sorry for the poor explanation. I was trying to use some of the amazing functionality here (loss and accuracy classes) without using your trainer. And I totally missed the below point in your docs that there is no validation step in the trainer. And that clears my question, ill close this. Thanks a lot!

> dataset: The dataset you want to train on. Note that training methods do not perform validation, so do not pass in your validation or test set.
